From 1ab8783e716eb5ae2fb44b06a2db16b9fb91dad9 Mon Sep 17 00:00:00 2001
From: Laurent Vivier <lvivier@redhat.com>
Date: Fri, 14 Aug 2020 16:02:49 -0400
Subject: [PATCH 2/4] redhat: update pseries-rhel8.2.0 machine type

RH-Author: Laurent Vivier <lvivier@redhat.com>
Message-id: <20200814160249.217753-3-lvivier@redhat.com>
Patchwork-id: 98156
O-Subject: [RHEL-AV-8.3.0 qemu-kvm PATCH 2/2] redhat: update pseries-rhel8.2.0 machine type
Bugzilla: 1843348
RH-Acked-by: Dr. David Alan Gilbert <dgilbert@redhat.com>
RH-Acked-by: Cornelia Huck <cohuck@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

v5.1.0-rc1 has modified the default state of pseries-5.1 and
introduced a new entry in pseries-5.0 machine type:
  a6030d7e0b35 ("spapr: Add a new level of NUMA for GPUs")

Add this entry to pseries-rhel8.2.0 as the default state has
also changed for pseries-rhel8.3.0

Signed-off-by: Laurent Vivier <lvivier@redhat.com>
Signed-off-by: Danilo C. L. de Paula <ddepaula@redhat.com>
---
 hw/ppc/spapr.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/hw/ppc/spapr.c b/hw/ppc/spapr.c
index 1c367a2367..5e3964326d 100644
--- a/hw/ppc/spapr.c
+++ b/hw/ppc/spapr.c
@@ -5019,10 +5019,15 @@ DEFINE_SPAPR_MACHINE(rhel830, "rhel8.3.0", true);
 static void spapr_machine_rhel820_class_options(MachineClass *mc)
 {
     SpaprMachineClass *smc = SPAPR_MACHINE_CLASS(mc);
+    /* from pseries-5.0 */
+    static GlobalProperty compat[] = {
+        { TYPE_SPAPR_PCI_HOST_BRIDGE, "pre-5.1-associativity", "on" },
+    };
 
     spapr_machine_rhel830_class_options(mc);
     compat_props_add(mc->compat_props, hw_compat_rhel_8_2,
                      hw_compat_rhel_8_2_len);
+    compat_props_add(mc->compat_props, compat, G_N_ELEMENTS(compat));
 
     /* from pseries-4.2 */
     smc->default_caps.caps[SPAPR_CAP_FWNMI] = SPAPR_CAP_OFF;
@@ -5031,6 +5036,7 @@ static void spapr_machine_rhel820_class_options(MachineClass *mc)
 
     /* from pseries-5.0 */
     mc->numa_mem_supported = true;
+    smc->pre_5_1_assoc_refpoints = true;
 }
 
 DEFINE_SPAPR_MACHINE(rhel820, "rhel8.2.0", false);
-- 
2.27.0

